File: 351003/Kisel/JPA/RV_Kisel_lab2_Task320/Controllers/NewsController.cs
using Microsoft.AspNetCore.Mvc;
using RV_Kisel_lab2_Task320.Models.Dtos;
using RV_Kisel_lab2_Task320.Services;

namespace RV_Kisel_lab2_Task320.Controllers;

[ApiController]
[Route("api/v1.0/news")] // Прямой путь, чтобы тесты его нашли
public class NewsController : ControllerBase
{
    private readonly INewsService _service;
    public NewsController(INewsService service) => _service = service;

    [HttpGet]
    public async Task<ActionResult> GetAll() => Ok(await _service.GetAllAsync());

    [HttpGet("{id}")]
    public async Task<ActionResult> GetById(int id)
    {
        var n = await _service.GetByIdAsync(id);
        if (n == null) return NotFound(new ErrorResponse { ErrorMessage = "Not Found", ErrorCode = "40401" });
        return Ok(n);
    }

    [HttpPost]
    public async Task<ActionResult> Create(NewsDto dto)
    {
        if (await _service.ExistsByTitleAsync(dto.Title))
            return StatusCode(403, new ErrorResponse { ErrorMessage = "Duplicate title", ErrorCode = "40301" });

        var result = await _service.CreateAsync(dto);
        return CreatedAtAction(nameof(GetById), new { id = result.Id }, result);
    }

    [HttpPut]
    public async Task<ActionResult> Update(NewsDto dto)
    {
        await _service.UpdateAsync(dto.Id, dto);
        return Ok(dto);
    }

    [HttpDelete("{id}")]
    public async Task<IActionResult> Delete(int id)
    {
        var existingCreator = await _service.GetByIdAsync(id);
        if (existingCreator == null) 
        {
            return BadRequest(new ErrorResponse { ErrorMessage = "Creator not found", ErrorCode = "40003" });
        }
        await _service.DeleteAsync(id);
        return NoContent();
    }
}

File: 351004/Leshok/lab3/publisher/src/controllers/NewsController.js
const NewsService = require('../services/NewsService');
const { NewsRequestDto } = require('../dtos/NewsDto');

class NewsController {
  async create(req, res, next) {
    try {
      const dto = new NewsRequestDto(req.body.creatorId, req.body.title, req.body.content);
      const result = await NewsService.create(dto);
      res.status(201).json(result);
    } catch (err) { next(err); }
  }

  async update(req, res, next) {
    try {
      const dto = new NewsRequestDto(req.body.creatorId, req.body.title, req.body.content);
      const result = await NewsService.update(req.params.id, dto);
      res.status(200).json(result);
    } catch (err) { next(err); }
  }

  async getOne(req, res, next) {
    try {
      const result = await NewsService.getById(req.params.id);
      res.status(200).json(result);
    } catch (err) { next(err); }
  }

  async getAll(req, res, next) {
    try {
      const result = await NewsService.getAll();
      res.status(200).json(result);
    } catch (err) { next(err); }
  }

  async delete(req, res, next) {
    try {
      await NewsService.delete(req.params.id);
      res.status(204).send();
    } catch (err) { next(err); }
  }

  async getCreatorByNewsId(req, res, next) {
    try {
      const result = await NewsService.getCreatorByNewsId(req.params.id);
      res.status(200).json(result);
    } catch (err) { next(err); }
  }

  async getStickersByNewsId(req, res, next) {
    try {
      const result = await NewsService.getStickersByNewsId(req.params.id);
      res.status(200).json(result);
    } catch (err) { next(err); }
  }
}

module.exports = new NewsController();